shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/response/PredictionsByStopJoinResponse.kt

```diff
@@ -0,0 +1,81 @@
+package com.mbta.tid.mbta_app.model.response
+
+import com.mbta.tid.mbta_app.model.ObjectCollectionBuilder
+import com.mbta.tid.mbta_app.model.Prediction
+import com.mbta.tid.mbta_app.model.Trip
+import com.mbta.tid.mbta_app.model.Vehicle
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class PredictionsByStopJoinResponse(
+    @SerialName("predictions_by_stop") val predictionsByStop: Map<String, Map<String, Prediction>>,
+    val trips: Map<String, Trip>,
+    val vehicles: Map<String, Vehicle>
+) {
+
+    constructor(
+        objects: ObjectCollectionBuilder
+    ) : this(
+        objects.predictions.values
+            .groupBy { it.stopId }
+            .mapValues { predictions -> predictions.value.associateBy { it.id } },
+        objects.trips,
+        objects.vehicles
+    )
+
+    companion object {
```

Comment:
Both of these helper functions operate on a specific `PredictionsByStopJoinResponse`; why are they in the companion object?

Reply:
Fair question! I'll pull them out of the companion. I do still have a preference for keeping it functional though and returning a new `PredictionsByStopJoinResponse` rather than mutating the existing one.